src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/FunctionCallUtilities.cs

```diff
@@ -80,10 +109,10 @@ internal static partial class FunctionCallHelpers
     /// <param name="parameters">The dictionary of parameters.</param>
     /// <param name="options">A <see cref="JsonSerializerOptions"/> governing serialization.</param>
     /// <returns>A JSON encoding of the parameters.</returns>
-    public static string FormatFunctionParametersAsJson(IDictionary<string, object?>? parameters, JsonSerializerOptions? options = null)
+    public static string FormatFunctionParametersAsJsonString(IDictionary<string, object?>? parameters, JsonSerializerOptions? options = null)
```

Comment:
Does this need to be public? It's really just a call to JsonSerializer.Serialize.

Reply:
They could be removed. I suppose their benefit is that they give you ready access to a JsonSerializerContext containing `IDictionary<string, object?>` so that one doesn't need to be added in the service.

Reply:
I should add that these particular helpers uphold the invariant required by `AIFunctionFactory`: that object types must be deserialized as `JsonElement`. This isn't always a given and we expose configuration flags in JSO that change this. In my view this is further evidence that the current shape of AIFunction/FunctionXContent is a leaky abstraction -- a non-polymorphic approach eliminates the need for such helpers altogether.

Reply:
You were going to prototype the alternative you had in mind? It's hard to reason about possible deficiencies here without that.

Reply:
I can try to revive my old prototype assuming there's enough buy-in from the rest of the team.

Reply:
Removed the formatter helpers in https://github.com/dotnet/extensions/pull/5513/commits/514f846e359ca7abc283320e70c6ea6b17571537

Reply:
> I can try to revive my old prototype assuming there's enough buy-in from the rest of the team.

If it fixes more issues than it creates, we should certainly consider it.